x/poolmanager/protorev.go

```diff
@@ -4,88 +4,121 @@ import (
 	}
+
+	takerFeeForStakers := sdk.Coin{}
+	if err := takerFeeForStakers.Unmarshal(bz); err != nil {
+		return sdk.NewCoin(denom, osmomath.ZeroInt()), err
+	}
+
+	return takerFeeForStakers, nil
 }
 
-func (k Keeper) SetTakerFeeTrackerForStakers(ctx sdk.Context, takerFeeForStakers sdk.Coins) {
-	newTakerFeeForStakers := types.TrackedVolume{
-		Amount: takerFeeForStakers,
+// UpdateTakerFeeTrackerForStakersByDenom increases the take fee for stakers tracker for the specified denom by the specified amount.
+func (k Keeper) UpdateTakerFeeTrackerForStakersByDenom(ctx sdk.Context, denom string, increasedAmt osmomath.Int) error {
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.KeyTakerFeeStakersProtoRevArray)
+	key := types.GetKeyPrefixTakerFeeStakersProtoRevByDenom(denom)
+
+	takerFeeForStakers, _ := k.GetTakerFeeTrackerForStakersByDenom(ctx, denom)
+	takerFeeForStakers.Amount = takerFeeForStakers.Amount.Add(increasedAmt)
+	bz, err := takerFeeForStakers.Marshal()
+	if err != nil {
+		return err
 	}
-	osmoutils.MustSet(ctx.KVStore(k.storeKey), types.KeyTakerFeeStakersProtoRev, &newTakerFeeForStakers)
+
+	store.Set(key, bz)
+	return nil
 }
 
-func (k Keeper) SetTakerFeeTrackerForCommunityPool(ctx sdk.Context, takerFeeForCommunityPool sdk.Coins) {
-	newTakerFeeForCommunityPool := types.TrackedVolume{
-		Amount: takerFeeForCommunityPool,
+// GetTakerFeeTrackerForCommunityPool returns the taker fee for community pool tracker for all denoms that has been
+// collected since the accounting height.
+func (k Keeper) GetTakerFeeTrackerForCommunityPool(ctx sdk.Context) []sdk.Coin {
+	takerFeesForCommunityPool := make([]sdk.Coin, 0)
+
+	store := ctx.KVStore(k.storeKey)
+	iterator := sdk.KVStorePrefixIterator(store, types.KeyTakerFeeCommunityPoolProtoRevArray)
+
+	defer iterator.Close()
+	for ; iterator.Valid(); iterator.Next() {
+		bz := iterator.Value()
+		takerFeeForCommunityPool := sdk.Coin{}
+		if err := takerFeeForCommunityPool.Unmarshal(bz); err == nil {
+			takerFeesForCommunityPool = append(takerFeesForCommunityPool, takerFeeForCommunityPool)
+		}
 	}
-	osmoutils.MustSet(ctx.KVStore(k.storeKey), types.KeyTakerFeeCommunityPoolProtoRev, &newTakerFeeForCommunityPool)
+
+	return takerFeesForCommunityPool
 }
 
-func (k Keeper) GetTakerFeeTrackerForStakers(ctx sdk.Context) (currentTakerFeeForStakers sdk.Coins) {
-	var takerFeeForStakers types.TrackedVolume
-	takerFeeFound, err := osmoutils.Get(ctx.KVStore(k.storeKey), types.KeyTakerFeeStakersProtoRev, &takerFeeForStakers)
-	if err != nil {
-		// We can only encounter an error if a database or serialization errors occurs, so we panic here.
-		// Normally this would be handled by `osmoutils.MustGet`, but since we want to specifically use `osmoutils.Get`,
-		// we also have to manually panic here.
-		panic(err)
+// GetTakerFeeTrackerForCommunityPoolByDenom returns the taker fee for community pool tracker for the specified denom that has been
+// collected since the accounting height.
```

Comment:
I think it is useful to have context in the godoc that if nothing exists in store, we return zero coin.

I know that the code is really down below but IMO when navigating the codebase this is useful. As well as when treating methods as "black boxes"

Applies to other similar methods

Reply:
Also added this in line comment here https://github.com/osmosis-labs/osmosis/pull/7240/commits/ee425f046e3a7df31cdd297df7dbb6098cacf834